internal/service/users_test.go

```diff
@@ -214,3 +217,255 @@ func TestUsersService_UpdateUserInfo(t *testing.T) {
 		})
 	}
 }
+
+func TestUsersService_Signup(t *testing.T) {
+	cases := map[string]struct {
+		email string
+		input *SignupUserInput
+		// setupMocks func(context.Context, *repoMocks.MockUsers, *serviceMocks.MockIdGen)
+		setupMocks func(context.Context, *repoMocks.MockUsers)
+		checkError func(*testing.T, error)
+	}{
+		"success": {
+			email: "JognDoe@example.com",
+			input: &SignupUserInput{
+				Email:       "JognDoe@example.com",
+				Password:    "John*123#",
+				FirstName:   "John",
+				LastName:    "Doe",
+				DisplayName: "JohnnyD",
+			},
+			// setupMocks: func(ctx context.Context, mockUsers *repoMocks.MockUsers, mockgenId *serviceMocks.MockIdGen) {
+			setupMocks: func(ctx context.Context, mockUsers *repoMocks.MockUsers) {
+				mockUsers.EXPECT().GetByEmail(ctx, "JognDoe@example.com").Return(nil, nil).Times(1)
+				mockUsers.EXPECT().Insert(ctx, gomock.Any()).Return(nil).Times(1)
+			},
+			checkError: noError,
+		},
+		"user_already_exist": {
+			email: "JognDoe@example.com",
+			input: &SignupUserInput{
+				Email:       "JognDoe@example.com",
+				Password:    "John*123#",
+				FirstName:   "John",
+				LastName:    "Doe",
+				DisplayName: "JohnnyD",
+			},
+			setupMocks: func(ctx context.Context, mockUsers *repoMocks.MockUsers) {
+				u := &core.User{
+					Id:               "1111111",
```

Comment:
This test is correct, but may be a bit misleading, as emails are different. Please make the emails equal if possible

Reply:
Correction: wrong return value in mock makes this test invalid